src/functional_test_runner/lib/providers/provider_collection.js

```diff
@@ -2,7 +2,8 @@ import { loadTracer } from '../load_tracer';
 import { createAsyncInstance, isAsyncInstance } from './async_instance';
 
 export class ProviderCollection {
-  constructor(providers) {
+  constructor(log, providers) {
+    this._log = log;
     this._instances = new Map();
     this._providers = providers;
```

Comment:
the _private naming thing is a thing we do? is the point just to indicate to others that this is meant to be used in a private way?